sympy/printing/repr.py

```diff
@@ -166,7 +166,7 @@ def _print_Symbol(self, expr):
             d['dummy_index'] = expr.dummy_index
 
         if d == {}:
-            return "%s(%s)" % (expr.__class__.__name__, self._print(expr.name))
+            return "%s(%s)" % (expr.__class__.__name__, self._print(repr(expr.name)))
```

Comment:
You could use `%r` instead of repr here.

Reply:
```suggestion
            return "%s(%s)" % (expr.__class__.__name__, self._print(repr(expr.name)))
```
@oscarbenjamin I tried using %r but it gives the following output
``` 
     >>> s = Symbol(u'text')
     >>> print(srepr(s))
    Symbol('text')
```
But the expected output is following
```
     >>> s = Symbol(u'text')
    >>> print(srepr(s))
    Symbol(u'text')
```
which the above code is giving.